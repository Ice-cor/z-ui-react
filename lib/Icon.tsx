import React from 'react';
import './importIcons';
import './icon.scss';

interface IconProps {
    name: string;
    onClick?: React.MouseEventHandler<SVGElement>;
}

const Icon: React.FC<IconProps> = (props) => {
    return (
        <svg className={`zui-icon`} onClick={props.onClick}>
            <use xlinkHref={`#${props.name}`}></use>
        </svg>
    );
};

export default Icon;