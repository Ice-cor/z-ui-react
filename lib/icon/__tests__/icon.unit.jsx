import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../icon';
import {mount} from 'enzyme';

describe('icon', () => {
    it('这是一个icon', () => {
        const json = renderer.create(<Icon name="info" />).toJSON();
        expect(json).toMatchSnapshot();
    });
    it('onClick', () => {
        const component = mount(<Icon name="info" onClick={fn}/>)
        component.find()
    })
});
