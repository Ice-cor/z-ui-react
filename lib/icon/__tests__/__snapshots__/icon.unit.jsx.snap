// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon 这是一个icon 1`] = `
<svg
  className="zui-icon"
>
  <use
    xlinkHref="#info"
  />
</svg>
`;
